### PR TITLE
fix: fix mypy errors in rest.py

### LINKED ISSUE #1600: mypy test failing for rest.py in generated golden files
See output here https://github.com/googleapis/gapic-generator-python/actions/runs/4196535511/jobs/7277634221 from PR https://github.com/googleapis/gapic-generator-python/pull/1599

```
nox > mypy --explicit-package-bases google
google/cloud/asset_v1/services/asset_service/transports/rest.py:485: error: Dict entry 0 has incompatible type "str": "Dict[<nothing>, <nothing>]"; expected "str": "str"  [dict-item]
```

The issue also appears in https://github.com/googleapis/python-bigtable/actions/runs/4196460916/jobs/7277462654 in https://github.com/googleapis/python-bigtable/pull/734
```
Run nox -s mypy
nox > Running session mypy
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/mypy
nox > python -m pip install -e .
nox > python -m pip install mypy types-setuptools types-protobuf types-mock
nox > python -m pip install google-cloud-testutils
nox > mypy google/ tests/
google/cloud/bigtable_v2/services/bigtable/transports/rest.py:30: error: Library stubs not installed for "requests"  [import]
google/cloud/bigtable_admin_v2/services/bigtable_table_admin/transports/rest.py:31: error: Library stubs not installed for "requests"  [import]
google/cloud/bigtable_admin_v2/services/bigtable_table_admin/transports/rest.py:3152: error: Dict entry 0 has incompatible type "str": "Dict[<nothing>, <nothing>]"; expected "str": "str"  [dict-item]
google/cloud/bigtable_admin_v2/services/bigtable_table_admin/transports/rest.py:3251: error: Dict entry 0 has incompatible type "str": "Dict[<nothing>, <nothing>]"; expected "str": "str"  [dict-item]
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:31: error: Library stubs not installed for "requests"  [import]
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:31: note: Hint: "python3 -m pip install types-requests"
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:31: note: (or run "mypy --install-types" to install all missing stub packages)
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:31: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:[11](https://github.com/googleapis/python-bigtable/actions/runs/4196460916/jobs/7277462654#step:5:12)29: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:2[17](https://github.com/googleapis/python-bigtable/actions/runs/4196460916/jobs/7277462654#step:5:18)1: error: Dict entry 0 has incompatible type "str": "Dict[<nothing>, <nothing>]"; expected "str": "str"  [dict-item]
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:[22](https://github.com/googleapis/python-bigtable/actions/runs/4196460916/jobs/7277462654#step:5:23)73: error: Dict entry 0 has incompatible type "str": "Dict[<nothing>, <nothing>]"; expected "str": "str"  [dict-item]
google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/transports/rest.py:[26](https://github.com/googleapis/python-bigtable/actions/runs/4196460916/jobs/7277462654#step:5:27)43: error: Dict entry 0 has incompatible type "str": "Dict[<nothing>, <nothing>]"; expected "str": "str"  [dict-item]
Found 9 errors in 3 files (checked 91 source files)
nox > Command mypy google/ tests/ failed with exit code 1
nox > Session mypy failed.
```

### PR DESCRIPTION
**Pull request by parthea**
Fixes #1600

Also addresses mypy error in golden file `credentials/google/iam/credentials_v1/types/iamcredentials.py`
```
nox > mypy --explicit-package-bases google
google/iam/credentials_v1/types/iamcredentials.py:18: error: Name "proto" is not defined  [name-defined]
```
https://github.com/googleapis/gapic-generator-python/blob/main/tests/integration/goldens/credentials/google/iam/credentials_v1/types/iamcredentials.py#L18

### CODE REVIEW

**Review by Mariatta — APPROVED** [review]

File: gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2 [hunk 1]
```diff
@@ -269,7 +269,7 @@ class {{service.name}}RestTransport({{service.name}}Transport):
 
         {% if not method.client_streaming %}
         {% if method.input.required_fields %}
-        __REQUIRED_FIELDS_DEFAULT_VALUES: Dict[str, str] =  {
+        __REQUIRED_FIELDS_DEFAULT_VALUES: Dict[str, Any] =  {
```
Comment (vchudnov-g) [inline gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2:272]: If the whole purpose of `typing` is to provide type hints, then isn't the use of `Any` defeating the purpose? Is there a more narrow type hint we could provide? (I see that the error message in [1] expects as the map value the type `"Dict[<nothing>, <nothing>]"`)

[1] https://github.com/googleapis/python-bigtable/actions/runs/4196460916/jobs/7277462654

File: gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2 [hunk 2]
```diff
@@ -269,7 +269,7 @@ class {{service.name}}RestTransport({{service.name}}Transport):
 
         {% if not method.client_streaming %}
         {% if method.input.required_fields %}
-        __REQUIRED_FIELDS_DEFAULT_VALUES: Dict[str, str] =  {
+        __REQUIRED_FIELDS_DEFAULT_VALUES: Dict = {
```
Comment (vchudnov-g) [inline gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2:272]: But only checking for `Dict` does even less type checking than `Dict[str, Any]`, right, so what's the point? I was wondering whether we could have the `str` check on the key but also make the type check tighter than just an `Any`. Does that make sense.

The more I think about it, the more I wonder whether it's a sensible question. I guess each required field can be of any proto-derived type, and there may be multiple required fields, so at best we could check for the union of the types of the required fields in the request message. Hmmmm.... Maybe you're right, `Dict[str,Any]` is the most sensible. Thoughts?

Comment (vchudnov-g) [reply gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2:272]: Thought about it some more. I suggest reverting your last commit and going back to `Dict[str,Any]` (sorry!) unless there's any way to make the `Any` more specific, which I'm thinking there isn't without a lot more work for arguably small benefit.....

Comment (parthea) [reply gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2]: Done

**Review by vchudnov-g — COMMENTED** [review]
lgtm. Thanks!

**Pull request merged**